src/cross_section.F90

```diff
@@ -526,6 +577,192 @@ pure function find_energy_index(mat, E) result(i)
+      call multipole_eval_allocate()
+    end if
+
+    ! Locate us
+    iW = floor((sqrtE - sqrt(multipole % start_E))/multipole % spacing + ONE)
+
+    startw = multipole % w_start(iW)
+    startw_1 = startw - 1 ! This is an index shift parameter.
+    endw = multipole % w_end(iW)
+    startw_endw = endw - startw + 1
+
+    ! Fill in factors
+    if (startw <= endw) then
+      call fill_factors(multipole, sqrtE, sigT_factor, twophi, multipole % num_l)
+    end if
+
+    ! Generate some doppler broadening parameters
+
+    ! dopp_ecoef is inverse of dopp, divided by E, multiplied by sqrt(pi).
+    dopp = multipole % sqrtAWR / sqrtKT
+    dopp_ecoef = dopp * invE * SQRT_PI
+
+    sigT = ZERO
+    sigA = ZERO
+    sigF = ZERO
+    ! Evaluate linefit first
+
+    if(sqrtkT /= 0 .and. multipole % broaden_poly(iW) == 1) then ! Broaden the curvefit.
+      call broaden_n_polynomials(E, dopp, multipole % fit_order + 1, broadened_polynomials)
+
+      do iC = 1, multipole % fit_order+1
+        sigT = sigT + multipole % curvefit(FIT_T, iC, iW)*broadened_polynomials(iC)
+        sigA = sigA + multipole % curvefit(FIT_A, iC, iW)*broadened_polynomials(iC)
+        if (multipole % fissionable) then
+          sigF = sigF + multipole % curvefit(FIT_F, iC, iW)*broadened_polynomials(iC)
+        end if
+      end do
+    else ! Evaluate as if it were a polynomial
+      temp = invE
+      do iC = 1, multipole % fit_order+1
+
+        sigT = sigT + multipole % curvefit(FIT_T, iC, iW)*temp
+        sigA = sigA + multipole % curvefit(FIT_A, iC, iW)*temp
+        if (multipole % fissionable) then
+          sigF = sigF + multipole % curvefit(FIT_F, iC, iW)*temp
+        end if
+
+        temp = temp * sqrtE
+      end do
+    end if
+
+    ! Then get the poles we want and broaden them.
+
+    if (sqrtkT == ZERO) then
+      ! If at 0K, use asymptotic form.
+      do iP = startw, endw
+        psi_ki = -ONEI/(multipole % data(MP_EA, iP) - sqrtE)
+        c_temp = psi_ki/E
+        if (multipole % formalism == FORM_MLBW) then
+          sigT = sigT + real(multipole % data(MLBW_RT, iP) * c_temp * &
+                             sigT_factor(multipole % l_value(iP))) &
+                      + real(multipole % data(MLBW_RX, iP) * c_temp)
+          sigA = sigA + real(multipole % data(MLBW_RA, iP) * c_temp)
+          sigF = sigF + real(multipole % data(MLBW_RF, iP) * c_temp)
+        else if (multipole % formalism == FORM_RM) then
+          sigT = sigT + real(multipole % data(RM_RT, iP) * c_temp* &
+                             sigT_factor(multipole % l_value(iP)))
+          sigA = sigA + real(multipole % data(RM_RA, iP) * c_temp)
+          sigF = sigF + real(multipole % data(RM_RF, iP) * c_temp)
+        end if
+      end do
+    else
+      ! At temperature, use Faddeeva function-based form.
+      if(endw >= startw) then
+        do iP = startw, endw
+          Z = (sqrtE - multipole % data(MP_EA, iP)) * dopp
+          w_val = w(Z) * dopp_ecoef
```

Comment:
This is the only place `dopp_ecoef` is used.  I don't think it's worth packing into it's own variable.  I would recommend getting rid of `dopp_ecoef` and just stating plainly
`w_val = w(Z) * dopp * invE * SQRT_PI`